module/jobqueue/README.md

```diff
@@ -0,0 +1,63 @@
+## JobQueue Design Goal
+
+The jobqueue package implemented a reusable job queue system for async message processing.
+
+The most common use case is to work on each finalized block async.
+
+For instance, the verification node verify each finalized block. The block processing needs to be async because if not, the verification node might get overwhelmed by the large amount of blocks to be processed when they got finalized quickly during the period that the node went back online and catching up blocks from other peers.
+
+So the goal for the jobqueue system are:
+1. guarantee each finalized block will be processed eventually
+2. in the event of a crash failure, the worker can be scheduled so that no finalized block will be skipped.
+3. allow concurrent processing of multiple jobs (i.e, finalized blocks)
+4. the number of concurrent workers are configable so that the node won't get overwhelmed when too many jobs are created (i.e too many blocks are finalized in a short period of time)
```

Comment:
```suggestion
4. the number of concurrent workers are configable so that the node won't get overwhelmed by too many jobs being created (i.e. too many blocks being finalized in a short period of time)
```